src/cmd/ksh93/sh/fault.c

```diff
@@ -432,13 +433,14 @@ void sh_chktrap(Shell_t *shp) {
 // Exit the current scope and jump to an earlier one based on pp->mode.
 //
 void sh_exit_20120720(Shell_t *shp, int xno) {
-    struct checkpt *pp = (struct checkpt *)shp->jmplist;
-    int sig = 0;
     Sfio_t *pool;
+    int sig = 0;
+    struct checkpt *pp = (struct checkpt *)shp->jmplist;
+    assert(pp);
```

Comment:
Note that this `assert(pp)` isn't really needed. If `pp` is NULL the program should die when the dereference three lines below occurs. This is here solely to make it blindingly obvious we require it to be non-NULL even though the prior version of the code makes it appear it could be NULL at this juncture.